projboard/tests/test_client.py

```diff
@@ -113,3 +114,60 @@ def test_aboutpage(client):
     template_names = set(tmpl.origin.template_name for tmpl in response.templates)
     # And check if that 'about/about.html' is in the set.
     assert 'about/about.html' in template_names
+
+
+@pytest.fixture
+@pytest.mark.django_db
+def User2(client):
+    return User.get_user_by_nickname("User2")
+
+
+@pytest.mark.django_db
+def test_read_new_article(client, world_cup_article, User2):
+
+    # recieving world cup article that User 2 didn't view yet, plus getting num views of it
+    num_views1 = world_cup_article.num_of_views
+
+    # preparing the page route
+    route = '/article/{0}/{1}'.format(world_cup_article.id, User2.nickname)
+
+    # Send a GET request to the page
+    response = client.get(route)
+
+    assert response.status_code == 200
+    # Create a set of template names from the templates used in the response
+    template_names = set(tmpl.origin.template_name for tmpl in response.templates)
+    # And check if that show/read_article.html is in the set.
+    assert 'show/read_article.html' in template_names
+    # checking the result of the view return
+    assert response.context['article'] == world_cup_article
+    # getting num of views after viewing the article
+    num_views2 = response.context['article'].num_of_views
+    # checking that num views has been increased
+    assert num_views2 == num_views1 + 1
+
+
+@pytest.mark.django_db
+def test_read_old_article(client, math_article, User2):
```

Comment:
see above comment

Reply:
Still I do not understand the tests here

Reply:
done